src/components/FlowView.tsx

```diff
@@ -0,0 +1,95 @@
+ * to learn more. Once redirected back to the app, provide the deep link via the `deepLink` prop.
+ *
+ * **Android Setup**
+ *
+ *    - **IMPORTANT NOTE**: even though Application links are the recommended way to configure
+ *      deep links, some browsers, such as Opera, do not honor them and open the URLs inline.
+ *      It is possible to circumvent this issue by using a custom scheme, albeit less secure.
+ *
+ * - Beyond that, in order to use navigation / redirection based authentication,
+ * namely `Magic Link`, `OAuth (social)` and SSO, it's required to set up app links.
+ * App Links allow the application to receive navigation to specific URLs,
+ * instead of opening the browser. Follow the [Android official documentation](https://developer.android.com/training/app-links)
+ * to set up App link in your application. Once redirected back to the app, provide the deep link via the `deepLink` prop.
+ *
+ * - Finally, it is possible for users to authenticate using the Google account or accounts they are logged into
+ * on their Android devices. If you haven't already configured your app to support `Sign in with Google` you'll
+ * probably need to set up your [Google APIs console project](https://developer.android.com/identity/sign-in/credential-manager-siwg#set-google)
+ * for this. You should also configure an OAuth provider for Google in the in the [Descope console](https://app.descope.com/settings/authentication/social),
+ * with its `Grant Type` set to `Implicit`. Also note that the `Client ID` and
+ * `Client Secret` should be set to the values of your `Web application` OAuth client,
+ * rather than those from the `Android` OAuth client.
+ * For more details about configuring your app see the [Credential Manager documentation](https://developer.android.com/identity/sign-in/credential-manager).
+ * @param props iOS and Android specific flow options to set up the flow, a `deepLink` to continue redirect based authentication
+ * a set of callbacks when the Flow is `ready` to be presented, and finished in a `success` or `error` state.
+ * @returns The Descope FlowView component
+ */
+export default function FlowView(props: { flowOptions: FlowOptions; deepLink?: string; style?: ViewStyle; onReady?: () => unknown; onSuccess?: (jwtResponse: JWTResponse) => unknown; onError?: (error: string) => unknown }) {
+  const { onReady, onSuccess, onError } = props
+  const onReadyCb = useCallback(() => {
+    onReady?.()
+  }, [onReady])
+
+  const onSuccessHook = useCallback(
+    (event: any) => {
+      const rawResponse = JSON.parse(event.nativeEvent.response)
+      const jwtResponse = rawResponse as JWTResponse
+      jwtResponse.sessionJwt = jwtResponse.sessionJwt ?? rawResponse.sessionToken
+      jwtResponse.refreshJwt = jwtResponse.refreshJwt ?? rawResponse.refreshToken
+      onSuccess?.(jwtResponse)
+    },
+    [onSuccess],
+  )
+
+  const onErrorCb = useCallback(
+    (event: any) => {
+      onError?.(event.nativeEvent.error)
```

Comment:
on browser, error contains a struct with error message, error code and error description, how this is mapped to a single error string?

Reply:
Currently it's the native representation. I can transfer the components like you mention - good idea.